### ISSUE TITLE
Woodpecker creates empty pipelines

### ISSUE THREAD

**Issue by xoxys** [woodpecker-ci/woodpecker#2901]
### Component

server

### Describe the bug

Woodpecker creates empty pipelines that never start because they are empty:

![image](https://github.com/woodpecker-ci/woodpecker/assets/3391958/6ee2b47d-d4dd-4058-8a04-11b0ab1bca52)
![image](https://github.com/woodpecker-ci/woodpecker/assets/3391958/67cf05d8-67d6-47d1-8c57-602e4d24968c)

Issue occurs in `next-621eed6205` while reverting to `next-111a0b4ea5` fixed it.

### System Info

```shell
next-621eed6205
```

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't [already an issue](https://github.com/woodpecker-ci/woodpecker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Checked that the bug isn't fixed in the `next` version already [https://woodpecker-ci.org/faq#which-version-of-woodpecker-should-i-use]
- [X] Check that this is a concrete bug. For Q&A join our [Discord Chat Server](https://discord.gg/fcMQqSMXJy) or the [Matrix room](https://matrix.to/#/#woodpecker:matrix.org).

**Comment by xoxys** [comment]
As you can see in the screenshot for the PR event on the same commit a Pipeline is running. There is a repo config but WP creates an empty job for the renovate branch in that example.

**Comment by xoxys** [comment]
Can still reproduce this on latest next release, see https://ci.rknet.org/repos/5

![image](https://github.com/woodpecker-ci/woodpecker/assets/3391958/3c8901ef-8138-44c1-8b04-f742c434fc71)

It creates an empty pipeline to push events to any branch. The Woodpecker repo in the config has not changed. You can find the config here https://gitea.rknet.org/container/hugo/src/branch/main/.woodpecker This config should ignore any push action to none-default branches. For whatever reason, the push action triggers "something" which leads to an empty pipeline that never starts.

**Comment by xoxys** [comment]
> Can you check the logs? (Maybe also with debug logs)

Nothing in the regular logs. Will switch to debug later and check again. I'll also try to reproduce it with a local dev setup.

**Comment by xoxys** [comment]
I'm also able to reproduce this on a local dev setup with woodpecker-main + gitea 1.21

![image](https://github.com/woodpecker-ci/woodpecker/assets/3391958/7c96dc40-6f45-4160-b93e-4702a22ba598)

```
❯ tree .woodpecker/
.woodpecker/
└── demo.yml

1 directory, 1 file

❯ cat .woodpecker/demo.yml 
when:
  - event: [pull_request, tag]
  - event: push
    branch:
      - ${CI_REPO_DEFAULT_BRANCH}

steps:
  demo:
    image: 'alpine'
    commands:
      - echo 'Demo'
```

**Comment by xoxys** [comment]
Any idea what PR could have introduced this? As the latest working version for me is `next-621eed6205` I can try to use bisect to find the commit which has introduced the issue.

**Comment by xoxys** [comment]
Found it:

```
❯ git bisect good
849b02a433c375edc1a8ebc52bf38d8e13263ba3 is the first bad commit
commit 849b02a433c375edc1a8ebc52bf38d8e13263ba3
Author: qwerty287 <80460567+qwerty287@users.noreply.github.com>
Date:   Sun Nov 26 19:59:36 2023 +0100

    Fix pipeline-related environment (#2876)
    
    closes https://github.com/woodpecker-ci/woodpecker/issues/2672
    
    pipeline model must be persisted first to have some fields like `Number`
    and `ID`

 server/pipeline/create.go | 24 +++++++++++-------------
 1 file changed, 11 insertions(+), 13 deletions(-)
```

Related to https://github.com/woodpecker-ci/woodpecker/pull/2876

**Comment by xoxys** [comment]
Debug log while the empty pipeline is being created in the UI:

```
{"level":"error","error":"token signature is invalid: signature is invalid","time":"2023-12-07T20:32:39Z","caller":"/build/server/api/hook.go:173","message":"failure to parse token from hook for testorg/woodpecker-test"}
{"level":"debug","ip":"172.22.0.2","latency":1.15212,"method":"POST","path":"/api/hook","status":400,"time":"2023-12-07T20:32:39Z","user-agent":"Go-http-client/1.1","time":"2023-12-07T20:32:39Z","caller":"/build/server/router/middleware/logger.go:59"}
{"level":"debug","pipeline":"demo","time":"2023-12-07T20:32:39Z","caller":"/build/pipeline/stepBuilder.go:158","message":"Marked as skipped, does not match metadata"}
{"level":"debug","repo":"testorg/woodpecker-test","time":"2023-12-07T20:32:39Z","caller":"/build/server/pipeline/create.go:83","message":"ignoring hook: 'when' filters filtered out all steps"}
{"level":"debug","ip":"172.22.0.2","latency":29.213733,"method":"POST","path":"/api/hook","status":204,"time":"2023-12-07T20:32:39Z","user-agent":"Go-http-client/1.1","time":"2023-12-07T20:32:39Z","caller":"/build/server/router/middleware/logger.go:59"}
```
